### ISSUE TITLE
URL changed from https to http

### ISSUE THREAD

**Issue by jbmulligan** [tus/tus-js-client#185]
I am running tusd behind a reverse proxy that maps https (on port 3080) to http (on port 1080, tusd default).  My endpoint is defined to be https://my_server:3080/files/.  But I am getting a javascript console error from tus-js-client/dist/lib/upload.js (line 504):

Refused to connect to 'http://my_server:3080/files/e3977e2e17175f935be4f00da3946d13' because it violates the following Content Security Policy directive: "connect-src 'self' blob: https://my_server:3080/".

Somewhere, the https that I provided in the endpoint is getting changed to http!?  The tusd docs talk a little about proxy issues [here](https://github.com/tus/tusd/blob/master/docs/faq.md).  In particular, it says to provide the X-Forwarded-Host and X-Forwarded-Proto headers, I am wondering if that may be the problem.  The host is OK but the proxy and tusd are running on the same host.  I am running a small little proxy server using nodejs and http-proxy, as described [here](https://github.com/http-party/node-http-proxy#using-https).  I don't know where to set something to provide the headers.  The uploads object has a headers option, described as "an object with custom header values used in all requests" but I can't find any documenation or examples showing how that should be used.  Can anyone enlighten me?

One other point that may be important:  I am using tus-js-client in conjunction with Uppy.  The endpoint is specified in two places in my code:  in the initialization of the Uppy dashboard, and again in a direct call to tus.Upload that I am making in order to also upload the thumbnail images.  The thumbnails are successfully uploaded, the error is thrown when Uppy tries to upload the main files.  So maybe the problem is with my usage of Uppy?

In summary, I am not sure if this forum is the right place for this problem, but I am starting here because this is where the error is thrown from.

Thanks in advance, Jeff

**Comment by jbmulligan** [comment]
SOLVED!  The problem was a missing xfwd: true option in creation of the http-proxy server.
Ack!

**Comment by Acconut** [comment]
Glad to read that you found the problem!
